### PR TITLE
chore: apply codacy fixes

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

We use Codacy to statically analyse the code and find vulnerabilities.

### CODE REVIEW

File: clients/algoliasearch-client-go/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/algolia/algoliasearch-client-go/v4
 
-go 1.21
+go 1.22
```
Comment (shortcuts) [inline clients/algoliasearch-client-go/go.mod:3]: 23 even, no?

**Review by shortcuts — COMMENTED** [review]
Linter as a service

File: clients/algoliasearch-client-go/go.mod (hunk 1 above)
Comment (millotp) [reply clients/algoliasearch-client-go/go.mod:3]: Can people using go1.22 still use this lib if we use 1.23 ?

Comment (shortcuts) [reply clients/algoliasearch-client-go/go.mod:3]: I think so yes, as long as we don't depend on std introduced in 1.21+

I guess we should add multi version testing to Go too D:

Comment (millotp) [reply clients/algoliasearch-client-go/go.mod:3]: Just tried it with the CTS: `go: module ../../../clients/algoliasearch-client-go requires go >= 1.23 (running go 1.22.8)`

Comment (millotp) [reply clients/algoliasearch-client-go/go.mod:3]: we should stick to a low version

Comment (shortcuts) [reply clients/algoliasearch-client-go/go.mod:3]: oki nice then let's keep 1.21 because that's what we settled on at the major release

**Review by shortcuts — COMMENTED** [review]
looks cleaaaan, not sure about Go

**Review by shortcuts — APPROVED** [review]
gégé

**Pull request merged**